narwhal/consensus/src/consensus.rs

```diff
@@ -31,6 +35,114 @@ pub mod consensus_tests;
 /// The representation of the DAG in memory.
 pub type Dag = BTreeMap<Round, HashMap<AuthorityIdentifier, (CertificateDigest, Certificate)>>;
 
+#[derive(Default)]
+pub struct LeaderSwapTable {
```

Comment:
It would be useful to leave a comment, about how `good_nodes` and `bad_nodes` are defined, and how they relate to the committee as a whole.